src/Solver/Dynamic/Ida.cpp

```diff
@@ -110,48 +116,113 @@ namespace AnalysisManager
       return this->configureLinearSolver();
     }
 
+    /**
+     * @brief Configure the linear solver
+     *
+     * @note This currently uses pre-processor directives to set dense or sparse
+     * linear solvers
+     *
+     * @tparam ScalarT
+     * @tparam IdxT
+     */
     template <class ScalarT, typename IdxT>
     int Ida<ScalarT, IdxT>::configureLinearSolver()
     {
       int retval = 0;
+
+/// Todo - Implement a cleaner way to handle the Sparse versus Dense solvers
+#ifdef GRIDKIT_ENABLE_SUNDIALS_SPARSE
+      if (model_->hasJacobian())
+      {
+        this->configureLinearSolverSparse();
+      }
+      else
+      {
+        this->configureLinearSolverDense();
+      }
+#else
```

Comment:
These ifs and `#ifdef` can be simplified slightly by moving the `#ifdef` inside the if block so the else case doesn't have to be duplicated.